python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+                               mask_files=maskFiles, gains=gains)
+            del trapTaskDataId
+
+        ################################
+        ################################
+        # Run the CTE task
+        if self.config.doCTE:
+            self.log.info("Starting CTE task")
+            cteTaskDataId = {'run': run, 'testType': 'SFLAT_500', 'imageType': 'FLAT'}
+            for ccd in ccds:
+                if 'SFLAT_500' not in testTypes:
+                    msg = "No superflats found. Available data: %s"%testTypes
+                    if self.config.requireAllEOTests:
+                        raise RuntimeError(msg)
+                    else:
+                        self.log.warn(msg + "\nSkipping CTE task")
+                        break
+                sflatFilenames = [butler.get('raw_filename', dataId={'visit': visit,
+                                                                     'ccd': ccd})[0][:-3]
+                                  for visit in butler.queryMetadata('raw', ['visit'], dataId=cteTaskDataId)]
+                self.log.trace("CTETask: Processing %s with %s files"%(ccd, len(sflatFilenames)))
+                maskFiles = self._getMaskFiles(self.config.eotestOutputPath, ccd)
+                self.cte.run(sensor_id=ccd, superflat_files=sflatFilenames, mask_files=maskFiles)
+            del cteTaskDataId
+
+        ################################
+        ################################
+        # Run the flatPair task
+        if self.config.doFlatPair:
+            self.log.info("Starting flatPair task")
+            flatPairDataId = {'run': run, 'testType': 'FLAT', 'imageType': 'FLAT'}
+            for ccd in ccds:
+                if 'FLAT' not in testTypes:
+                    msg = "No dataset for flat_pairs found. Available data: %s"%testTypes
+                    if self.config.requireAllEOTests:
+                        raise RuntimeError(msg)
+                    else:
+                        self.log.warn(msg + "\nSkipping flatPair task")
+                        break
+                flatPairFilenames = [butler.get('raw_filename', dataId={'visit': visit,
+                                                'ccd': ccd})[0][:-3]
+                                     for visit in butler.queryMetadata('raw', ['visit'],
+                                                                       dataId=flatPairDataId)]
+                # Note that eotest needs the original filenames as that is the only place the flat
+                # pair-number is kept, so we have to resolve links and pass in the *original* paths here :(
+                # Also, there is no "flat-pair" test type, so all FLAT/FLAT imType/testType will appear here
+                # so we need to filter these for only the pair acquisitions (as the eotest code looks like it
+                # isn't totally thorough on rejecting the wrong types of data here)
```

Comment:
Can the camera team be persuaded to write the required data in the header?

You should explain what you mean by the "original" filename. I think you mean the filename written by the data acquisition system, which gets changed when we ingest into the data repo?

Reply:
Why do you need to rely on "flat1" and "flat2"? Can't you figure out pairs using the exposure time or something like that?

Reply:
The camera team can't change anything, they are in "production mode" and nothing will be changed at this point.

I have improved the explanation of what is meant by "original filename".

You could indeed work it out like that, but that is not how the 3rd party code works, and this code just exists to feed those tasks.